frontend/src/app/genres/genres.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { IGenre } from '../shared/models/IGenre';
+import { MatTableModule } from '@angular/material/table';
+import { HttpClient } from '@angular/common/http';
+import { ActivatedRoute, Router } from '@angular/router';
+import { GenresService } from '../shared/services/genres.service';
+import { Observable } from 'rxjs';
+import { GenreUpdateComponent } from '../genre-update/genre-update.component';
+@Component({
+  selector: 'app-genres',
+  templateUrl: './genres.component.html',
+  styleUrls: ['./genres.component.scss']
+})
+export class GenresComponent implements OnInit {
+  @ViewChild(GenreUpdateComponent) update!: GenreUpdateComponent;
+  genres: IGenre[] = [];
+  genreId: number | undefined;
+  displayedColumns: string[] = ['id', 'name', 'update', 'delete'];
+  edit: boolean = false;
+  add: boolean = false;
+  constructor(private router: Router, private _httpClient: HttpClient, private genresService: GenresService, private activatedRoute: ActivatedRoute) { }
+
+  ngOnInit(): void {
+    this.getData();
+  }
+
+  getData() {
+    this.genresService.getGenres().subscribe({
+      next: genres => {
+        this.genres = genres;
+        
+      },
+      error: err => {
+        console.log('getSensors subscribe -> error notification');
+      },
+      complete() {
+        
+      }
+    })
+  }
+  onUpdateClick(genreId: number) {
+    if(this.add==true){
```

Comment:
`if(this.add)`
Anything "truthy" (including boolean true, of course) doesn't require == true, it can be simply used as is in the if condition. But beware when using pure Javascript, since it can and will cause headaches.